### PR TITLE
CDAP-14603 add namespace to admin methods

### PR DESCRIPTION
**Pull request by albertshau**
Adding namespace as a parameter to ArtifactManager and
DatasetManager interfaces, as well as a namespace check to the
Admin interface.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT6673

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java [hunk 1]
```diff
@@ -128,16 +129,19 @@ public DynamicDatasetCache get() {
   private static final class AdminProvider implements Provider<Admin> {
     private final DatasetFramework dsFramework;
     private final SecureStoreManager secureStoreManager;
+    private final NamespaceAdmin namespaceAdmin;
 
     @Inject
-    private AdminProvider(DatasetFramework dsFramework, SecureStoreManager secureStoreManager) {
+    private AdminProvider(DatasetFramework dsFramework, SecureStoreManager secureStoreManager,
+                          NamespaceAdmin namespaceAdmin) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java]: Do you need the `NamespaceAdmin`? Can you use `NamespaceQueryAdmin` instead? I remember there was some complicate circular guice module issue when `NamespaceAdmin` is being used.

Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java]: Nope don't need the full NamespaceAdmin, didn't know NamespaceQueryAdmin existed. Will change.

File: cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactManager.java [hunk 2]
```diff
@@ -35,6 +35,14 @@
    */
   List<ArtifactInfo> listArtifacts() throws IOException;
 
+  /**
+   * Get the list of artifacts in the repository in the specified namespace and system namespace
+   *
+   * @return {@link List<ArtifactInfo>} list of artifact info
```
Comment (anwar6953) [inline cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactManager.java]: This would link to what? The List interface?

Comment (albertshau) [reply cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactManager.java]: i think we can remove this, was from copy-paste of the existing one.

Comment (albertshau) [reply cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactManager.java]: I'm missing a param in the javadoc, will fix this

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java [hunk 3]
```diff
@@ -158,4 +164,13 @@ public void deleteTopic(final String topic) throws TopicNotFoundException, IOExc
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public boolean namespaceExists(String namespace) throws IOException {
+    try {
+      return namespaceQueryAdmin.exists(new NamespaceId(namespace));
+    } catch (Exception e) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java:173]: It's better to throw `IOException` as is if the exception `e` is already an `IOException`.

```
} catch (IOException e) {
  throw e;
} catch (Exception e) {
  throw new IOException(e);
}

**Review by chtyim — APPROVED** [review]
LGTM

**Pull request merged**